types/_utils.d.ts

```diff
@@ -0,0 +1,59 @@
+import { ReactElement } from 'react'
+import ReactPlayer from './index'
+
+export interface SourceProps {
+  src: string
+  type: string
+}
+
+export interface BaseReactPlayerProps {
+  url?: string | string[] | SourceProps[] | MediaStream
+  playing?: boolean
+  loop?: boolean
+  controls?: boolean
+  volume?: number
+  muted?: boolean
+  playbackRate?: number
+  width?: string | number
+  height?: string | number
+  style?: Object
+  progressInterval?: number
+  playsinline?: boolean
+  playIcon?: ReactElement
+  pip?: boolean
+  stopOnUnmount?: boolean
+  light?: boolean | string
+  wrapper?: any
+  onReady?: (player: ReactPlayer) => void
+  onStart?: () => void
+  onPlay?: () => void
+  onPause?: () => void
+  onBuffer?: () => void
+  onBufferEnd?: () => void
+  onEnded?: () => void
+  onEnablePIP?: () => void
+  onDisablePIP?: () => void
+  onError?: (error: any, data?: any, hlsInstance?: any, hlsGlobal?: any) => void
+  onDuration?: (duration: number) => void
+  onSeek?: (seconds: number) => void
+  onProgress?: (state: {
+    played: number
+    playedSeconds: number
+    loaded: number
+    loadedSeconds: number
+  }) => void
+  [otherProps: string]: any
+}
+
+export class BaseReactPlayer {
```

Comment:
In my opinion, instead of repeating all properties of class in every extended player is redundant. You can use generics to remove this redundancy.
You should extend your base class from React.Component and use generics to implement extends base props types.
Following syntax:
```js
export default class BaseReactPlayer<T extends BaseReactPlayerProps> extends React.Component<T, any> {
   // all properties
}
```
Now in every player types use this exported class like this:
```js
export default class DailyMotionPlayer extends BaseReactPlayer<DailyMotionPlayerProps> {
  // any extended properties for dailymotion only.
}
```
In this case, you will not have to extend DailyMotionPlayerProps from BaseReactPlayerProps because generics will extends thos automatically.

Reply:
Thank you. I didn't know how to do it correctly